### PR TITLE
Minor code updates from missed open comments from #36 to the migratedown feature

### PR DESCRIPTION
**Pull request by venkat-iblox**
This PR addressed open comments from PR #36 

- renamed dirtyStateHandler to dirtyStateConfig
- used IsDirtyHandlingEnabled() in the `runMigrations` function instead of explicitly checking if the dirty state handling is enabled
- updated comments to use destination dir
- used https://pkg.go.dev/testing#T.TempDir in the UTs added

### CODE REVIEW

**Review by daniel-garcia — APPROVED** [review]
The code changes look fine. The commit messages and PR title needs to be descriptive of what is being changed.

File: migrate.go [hunk 1]
```diff
@@ -1087,7 +1086,7 @@ func (m *Migrate) logErr(err error) {
 func (m *Migrate) handleDirtyState() error {
 	// Perform the following actions when the database state is dirty
 	/*
-	   1. Update the source driver to read the migrations from the mounted volume
+	   1. Update the source driver to read the migrations from the destination path
```
Comment (akleinib) [inline migrate.go:1089]: `mounted volume` is used line 1094.

**Comment by akleinib** [comment]
All my comments have been addressed However it doesn't seem that changes related to Docker and build pertain to this PR.

**Comment by venkat-iblox** [comment]
> All my comments have been addressed However it doesn't seem that changes related to Docker and build pertain to this PR.

Thank you, I have removed the files for building a local image for testing. We probably should add a few of these paths into `.gitignore`.

**Review by akleinib — APPROVED** [review]

**Pull request merged**